edit: reworks the enhanced workflow doc single page (#1603)
mia-platform/documentation: docs/development_suite/set-up-infrastructure/revisions-and-versions.md

:::

#### Snaphot review 

On the revision history, it is possible to click on a snapshot ID to access its configuration and review and edit it using the Design section.

![load-snapshot-link](img/load-snapshot-link.png)

Once a snapshot is loaded, a warning message will appear as an overlay at the bottom with a button to exit the snapshot and return to the latest revision configuration.

![snapshot-view-on-design](img/snapshot-view-on-design.png)

From the revision status popover it is also possible to:

- exit the snapshot
- discard snapshot changes without exiting the snapshot review
- save snapshot changes to a new revision

![snapshot-view-actions](img/snapshot-view-actions.png)

### Default revision configuration

In the [Project Settings](/console/project-configuration/project-settings.md) section, if the Enhanced Workflow is enabled, it is possible to select one of the already existing revisions as **default revision**.

:::info
If no default revision is specified, the Console will use `main` as the default revision.
This revision will have an empty configuration if no other `main` branch was previously created on the Project.
:::

## Version History

> As the Revision History, there is also the Version History that list all the versions created on the Project.

Reviewer: ```suggestion
Similarly to the Revision History, the Version History lists all the versions created on the Project.
```

<div style={{display: 'flex', justifyContent: 'center'}}>

![Version History Button](img/version-history-button.png)

</div>

> Open the Version History, selecting any version in the Design Area and clicking the related icon in the branch action popover as shown in the image above.

Reviewer: ```suggestion
To open the Version History, select any version in the Design Area and click the related icon in the branch action popover as shown in the image above.
```

![Version History modal](img/version-history-modal.png)

> The table shows the name of each version with other useful info as the author name, the creation date and the description. Expanding the collapsed row of each versions it is possible to read also its release note, if provided and clicking on the version name you can load directly the version configuration in the Design area.

Reviewer: ```suggestion
The table shows the name of each version with other useful info such as the author's name, the creation date and the description. Expanding the collapsed row of each version makes it possible to read its release note (if provided), and by clicking on the version name you can load the version configuration directly in the Design area.
```

---
id: revisions-and-versions
title: Manage Revisions and Versions
sidebar_label: Manage Revisions and Versions
---

> In the project with the Enhanced Workflow the `Revisions` and `Versions` replacing the concept of `Branches` and `Tags` because the Console will not rely on the typical Git-based instruments anymore and will instead adopt a **new configuration management system**.

Reviewer: ```suggestion
In Projects adopting the Enhanced Workflow, `Revisions` and `Versions` replace the concept of `Branches` and `Tags`. The Console will not rely on the typical Git-based instruments anymore and will instead adopt a **new configuration management system**.
```

Developers can manage *revisions* and *versions* from the dedicated sections inside the Project Overview area.

## Revisions Section

 Revisions can be accessed and deleted from the Revisions management page:

![Revisions management page](img/revisions-overview.png)

## Versions Section

> Versions works as immutable snapshots of the configuration in a particular moment, with their description and release note, can be viewed and deleted through the versions management page:

Reviewer: ```suggestion
Versions work as immutable snapshots of the configuration in a particular moment, with their description and release note. They can be viewed and deleted through the versions management page:
```

It contains the list of snapshots of the Project configuration on the current revision or version, created each time it has been saved.  
It also includes useful information such as the snapshot date, author, associated message, and any tags associated with it.
Moreover, the right-side button allows the user to visualize changes between the selected snapshot and the previous one, in order to better understand the configuration's evolution over time.

![Diff Editor on Revision History modal](img/diff-editor-on-revision-history.png)

:::tip
From the revision history you can restore previous changes using the specific option in the contextual menu.

![restore button](img/revision-history-restore-changes-button.png)
:::

#### Snaphot review 

On the revision history, it is possible to click on a snapshot ID to access its configuration and review and edit it using the Design section.

![load-snapshot-link](img/load-snapshot-link.png)

Once a snapshot is loaded, a warning message will appear as an overlay at the bottom with a button to exit the snapshot and return to the latest revision configuration.

![snapshot-view-on-design](img/snapshot-view-on-design.png)

> It is also possible to leave the snapshot review also in the revision status popover or to save changes in a new revision or discard them.

Reviewer: ```suggestion
From the revision status popover it is also possible to:

- exit the snapshot
- discard snapshot changes without exiting the snapshot review
- save snapshot changes to a new revision
```

![snapshot-view-actions](img/snapshot-view-actions.png)

### Default revision configuration

In the [Project Settings](/console/project-configuration/project-settings.md) section, if the Enhanced Workflow is enabled, it is possible to select one of the already existing revisions as **default revision**.

:::info
If no default revision is specified, the Console will use `main` as the default revision.

> This revision will be an empty configuration if no other `main` branch was previously created on the Project.

Reviewer: ```suggestion
This revision will have an empty configuration if no other `main` branch was previously created on the Project.
```
